### ISSUE TITLE
Backup and Restore function

### ISSUE THREAD

**Issue by stoneubi** [KIMB-technologies/Radio-API#27]
It would be great to have an backup and restore function for the entire system configuration and radios.

E.g. I need to transfer the actual docker image from on server to another server.

In this case it would we good to open the GUI and to have a button called backup. This backup function should export the existing system and all radios including all stations to one file, which could the saved at the local pc. After that I would like to install the new docker image at another server and once the docker image is up and running, I would like to open the GUI and import the backup file. This should overwrite the whole system and should import all radios with all radio stations assigned to the radios.

At the last step I would change the dns settings to address the new ip and viola the transfer has been finished. If I now switch on the radio everything would be the same and no changes have to be done on the radio.

This function could also be used for regular backups 👍😃

Maybe you could implement this useful function.

Thank you so much
Stoneubi

**Comment by Himanshusoni9** [comment]
One More issue : On  Solr Collection Data Backup Based on Condition/Data Filter.(There is no provision for that . )

Because SOLR BACKUP API with Query is not working
//http://localhost:8983/solr/admin/collections?action=RESTORE&name=myBackupName&location=C:\Users\DELL\Downloads\SOLR_BACKUP&collection=myCondCollection&query=text:cellphone

I am trying out to perform backups of our Solr data with a particular condition in mind.

To provide some context, let's say Solr collection consists of 100 records, among which 70 records contain the text "mobile," and the remaining 30 records contain the text "cellphone." my objective is to take a Solr collection/data backup that contains only the records the text "cellphone" – essentially, we want to create a backup file that reflects these 30 specific records only.

I would greatly appreciate it if you could share insights on the best practices or methods to achieve this selective backup based on a condition. If there are specific parameters or commands we should be utilizing, kindly provide the necessary guidance. Additionally, any documentation or references you could point us to would be immensely helpful.

Thank you in advance for your time and assistance. We value your expertise and look forward to implementing an efficient solution based on your recommendations.

**Comment by kimbtech** [comment]
Radio-API does not use Solr and thus does not provide any Solr API access. The comment seems to be out of context here, thus I will hide it.

**Comment by stoneubi** [comment]
Creating a whole user account management just for this would be a show stopper.

Being able to import and export the settings view the gui would be a helpful feature in my opinion.

Maybe it would help to implement something else to avoid a misuse of this feature. Imaging I would press the import button and I have to enter a code, which is only shown on the radio. So I have to be in front of the radio / I have to have access to the radio to be able to import the settings. This would make sure that only people how know what the do will overwrite the settings. Wood this be a simple method?

Thank you
Stoneubi

**Comment by kimbtech** [comment]
I added the feature in Version [2.7.2](https://github.com/KIMB-technologies/Radio-API/releases/tag/v2.7.2).

After the Update a token `CONF_IM_EXPORT_TOKEN` needs to be set in config. Using this token the Im- & Export interface can be used at `./gui/im-export.php`.

Relying on a token shown in a radio did not convince me, as there might be multiple radios and one user could overwrite the settings of another user. Thus, I added the token via the config, which can be interpreted as an admin password.
